Code/GraphMol/SmilesParse/catch_tests.cpp

```diff
@@ -254,6 +254,40 @@ TEST_CASE("github #2257: writing cxsmiles", "[smiles,cxsmiles]") {
     auto smi = MolToCXSmiles(*mol);
     CHECK(smi == "CN |$_AV:val1;val2$,atomProp:0.p2.v2:1.p1.v1|");
   }
+
+  SECTION("mol fragments1") {
+    auto mol = "Cl.OC |(1,0,0;0,.75,0.1;0,-.75,-0.1)|"_smiles;
+    REQUIRE(mol);
+    CHECK(mol->getNumConformers() == 1);
+
+    std::vector<int> atomsToUse = {1, 2};
+    auto smi = MolFragmentToCXSmiles(*mol, atomsToUse);
+    CHECK(smi == "CO |(0,-0.75,-0.1;0,0.75,0.1)|");
+  }
+  SECTION("mol fragments2") {
+    auto mol = "Cl.N1CC1C |atomProp:1.p2.v2:1.p1.v1:2.p2.v2:2.p1.v1|"_smiles;
+    REQUIRE(mol);
+    CHECK(mol->getNumAtoms() == 5);
+    CHECK(!mol->getAtomWithIdx(0)->hasProp("p1"));
+    CHECK(mol->getAtomWithIdx(1)->hasProp("p1"));
+    CHECK(mol->getAtomWithIdx(1)->getProp<std::string>("p1") == "v1");
+
+    std::vector<int> atomsToUse = {1, 2, 3, 4};
+    auto smi = MolFragmentToCXSmiles(*mol, atomsToUse);
+    CHECK(smi == "CC1CN1 |atomProp:2.p2.v2:2.p1.v1:3.p2.v2:3.p1.v1|");
```

Comment:
I think this is wrong.  N here is Cdx 1 originally and here is cdx 4. ( atom idx 1 and 3, respectively). However, there is no atom 4 in the cdx output, just 2 and 3.

Reply:
I think it's right. Here's the mapping:

| Original index | Output Index | Properties |
|--------------------|-------------------|---------------|
| N 1 | N 3 | `p2:v2` `p1:v1` |
| C 2 | C 2 | `p2:v2` `p1.v1` |

It's not the best example in the world since the properties are the same. I will update that.

Reply:
I just pushed a commit that makes the properties different. Better test, hopefully clearer